### DISCUSSION TITLE
How to fix sync command not syncing slash commands

### DISCUSSION THREAD

**Discussion by AJwilky** [Rapptz/discord.py#9498] [Q&A]
So I have developed a discord bot with slash commands but of course for them to be used in multiple servers that need to be synced globally. I have written this command to do just that

```@client.command()
async def sync(ctx):
    print("sync command")
    if ctx.author.id == my_id:
        await client.tree.sync()
        await ctx.send(f"Synced commands to the the servers!")
    else:
        await ctx.send('You must be the owner to use this command!')
```        

This commands follows through and does send the "Synced commands to the the servers" meaning that the tree.sync() functions supposedly worked. I am aware you are supposed to wait a few hours so I just left the script running overnight and my commands still did not appear in any other servers except for the server I previously synced using the function. Is this written correctly and is there any reason it is not syncing to other servers even though I let it wait for a few hours?

**Comment by Duued** [comment]
Hey there! If your slash command is set to a specific guild, you will need to have `tree.sync` specify the specific guild. Also make sure your cog(s) are loaded correctly. The "waiting a few hours" was mainly when they were first released, now slash commands sync pretty much instantly.

**Reply by AJwilky** [reply]
No I'm not trying to sync it to a specific guild. I'm trying to sync it globally throughout all the guilds the bot is in.

**Comment by LostLuma** [comment]
Without knowing more about your bot I'd assume they only appear in a single server because either the bot is missing the `applications.commands` scope in the other server, or you are expecting a guild command to show up elsewhere after syncing global commands. If neither of these is the case please post more of your code here.

**Reply by AJwilky** [reply]
So when I wrote this question the applications.commands scope was not used and I have added to the application, however upon adding the applications.commands function it is still not syncing the commands, I could send more code but I am not sure exactly what you would like to be sent.

Here is the client variable 
```
PREFIX = ";"

INTENTS = discord.Intents.all()

client = commands.Bot(intents=INTENTS, command_prefix=PREFIX)

```

**Reply by LostLuma** [reply]
I'd mostly be interested in the commands themselves and their registration, to ensure you're not using the [`@guilds` decorator](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.guilds) or using [`CommandTree.add_command`](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.CommandTree.add_command) to restrict which guilds the commands are for, but you can really check that yourself.

If that's not the case you could create a [gist](https://gist.github.com/) (or a GitHub repo if it's many files) with your code so we can look over it to spot the issue, because it's mostly guesswork to see which small thing is amiss otherwise.

**Reply by AJwilky** [reply]
I found the error, I forgot I had left the guild= in the actual commands meaning it was only loading to that guild, I just made a careless mistake but I noticed it due to this comment so thank you and my apologies.

**Reply by LostLuma** [reply]
np
